### ISSUE TITLE
No compatible CUDA device is available

### ISSUE THREAD

**Issue by rubinanoor9** [openmm/openmm#3781]
Hello OpenMM users!
I am facing a problem during script running in openmm. Whenever I have tried to run the openmm script, so at each time it gives me the error "OpenMMException: No compatible CUDA device is available"

But when I start again to run the openmm script again, it works perfectly.
I am very confused about this error and how to solve it. I have installed openmm by the following command:
"conda install -c conda-forge openmm cudatoolkit=10.2"

I am looking forward to hearing from you.
Thanks in advance
Regards
Rubina

**Comment by peastman** [comment]
Is it possible your GPU is set in exclusive mode?  If so, only one process can use CUDA at a time.  What's the output of `nvidia-smi`?

**Comment by rubinanoor9** [comment]
The output of nvidia-smi look like this,

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.36       Driver Version: 440.36       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 106...  Off  | 00000000:04:00.0  On |                  N/A |
| 40%   48C    P2    24W / 120W |    111MiB /  3018MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1648      G   /usr/bin/X                                    62MiB |
|    0      1999      G   /usr/bin/plasmashell                          42MiB |
|    0      2184      G   ...r/1001/plasmashellzYrNhC.7.slave-socket     2MiB |
+-----------------------------------------------------------------------------+

**Comment by peastman** [comment]
It's in default mode, so that should be ok.  Can you explain exactly what the behavior is?  I don't quite understand your description:

> Whenever I have tried to run the openmm script, so at each time it gives me the error "OpenMMException: No compatible CUDA device is available"
>
> But when I start again to run the openmm script again, it works perfectly.

In what cases does it work, and in what cases does it give the error?

**Comment by peastman** [comment]
Did you manage to figure this out?  I see there was never a reply to my last post.

**Comment by rubinanoor9** [comment]
No, this problem has not been solved. Actually, I have set the environment in such a manner:

platform = Platform.getPlatformByName('CUDA')
properties = {'CudaPrecision': 'mixed'}


When I run this script, it gives me errors as mentioned above, but sometimes using the same script doesn't make errors. It runs smoothly.

I do not understanding why it happens.
